drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -68,13 +68,12 @@ public ScenarioSimulationEditorPresenter() {
     }
 
     @Inject
-    public ScenarioSimulationEditorPresenter(final ScenarioSimulationView baseView,
-                                             final Caller<ScenarioSimulationService> service,
+    public ScenarioSimulationEditorPresenter(final Caller<ScenarioSimulationService> service,
                                              final ScenarioSimulationResourceType type,
                                              final ImportsWidgetPresenter importsWidget,
                                              final AsyncPackageDataModelOracleFactory oracleFactory) {
-        super(baseView);
-        this.view = baseView;
+        super();
+        this.baseView = newScenarioSimulationView();   // Indirection added for test-purpose
```

Comment:
You should use the super(baseView) to provide the view for base and then set the view for this class so that you don't need to later cast for the actual view.

Reply:
I've removed the injection of baseview to reduce CDI usage (as you mention on next comment). Then, I've add  "newScenarioSimulationView" for testing purposes.
I've just restored usage of ScenarioSimulationView view to avoid casting with

[DROOLS-2672] Create static Scenario grid based on Uberfire Wires Core Grids - Restoring usage of "ScenarioSimulationView view" property to avoid casting.